src/content/documentation/develop/best-practices-for-collecting-user-data-consents.md

```diff
@@ -190,7 +188,7 @@ The add-on is requesting both types of data collection. Please ensure the choice
 
   ![Mockup of a prompt that could be used when extension requires consent for processing personal and technical data.](/assets/img/documentation/develop/consent-mixed.jpg)
 ### Required data collection
-The extension requires personal or technical data collection to provide its functionality and cannot be used without. The consent decline option uninstalls the extension.
+The extension requires personal or technical data collection to provide its main functionality and cannot be used without. The consent decline option uninstalls the extension using [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).
```

Comment:
Are extensions required to uninstall themselves if the user opts out of required data collection? If not, we may want to clarify that here and in other parts of this document.

```suggestion
The extension requires personal or technical data collection to provide its main functionality and cannot be used without this data. The consent decline option should either disable all data collection or uninstall the extension using [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).
```

Reply:
> Are extensions required to uninstall themselves if the user opts out of required data collection
Yes, we require them to do that. Since the pretext to this is that the extension requires the said data to provide its main functionality, the assumption is that it cannot be disabled and warrants an uninstall